stdlib/public/core/GraphemeWalker.swift

```diff
@@ -0,0 +1,370 @@
+      }
+
+      return false
+
+    // GB9a
+    case (_, .spacingMark):
+      return false
+
+    // GB9b
+    case (.prepend, _):
+      return false
+
+    // GB11
+    case (.zwj, .extendedPictographic):
+      if state.isBackwards {
+        checkIfInEmojiSequence(&state)
+      }
+
+      if state.isInEmojiSequence {
+        return false
+      } else {
+        return true
+      }
+
+    // GB12 & GB13
+    case (.regionalIndicator, .regionalIndicator):
+      if state.isBackwards {
+        countRIs(&state)
+      }
+
+      state.shouldBreakRI.toggle()
+      return !state.shouldBreakRI
+
+    // GB999
+    default:
+      return true
+    }
+  }
+
+  // When walking backwards, it's impossible to know whether we were in an emoji
+  // sequence without walking further backwards. This walks the string backwards
+  // enough until we figure out whether or not to break our
+  // (.zwj, .extendedPictographic) question. For example:
+  //
+  // Scalar view #1:
+  //
+  //     [.control, .zwj, .extendedPictographic]
+  //                     ^
+  //                     | = To determine whether or not we break here, we need
+  //                         to see the previous scalar's grapheme property.
+  //          ^
+  //          | = This is neither .extendedPictographic nor .extend, thus we
+  //              were never in an emoji sequence, so break between the .zwj
+  //              and .extendedPictographic.
+  //
+  // Scalar view #2:
+  //
+  //     [.extendedPictographic, .zwj, .extendedPictographic]
+  //                                  ^
+  //                                  | = Same as above, move backwards one to
+  //                                      view the previous scalar's property.
+  //                ^
+  //                | = This is an .extendedPictographic, so this indicates that
+  //                    we are in an emoji sequence, so we should NOT break
+  //                    between the .zwj and .extendedPictographic.
+  //
+  // Scalar view #3:
+  //
+  //     [.extendedPictographic, .extend, .extend, .zwj, .extendedPictographic]
+  //                                                    ^
+  //                                                    | = Same as above
+  //                                         ^
+  //                                         | = This is an .extend which means
+  //                                             there is a potential emoji
+  //                                             sequence, walk further backwards
+  //                                             to find an .extendedPictographic.
+  //
+  //                               <-- = Another extend, go backwards more.
+  //                ^
+  //                | = We found our starting .extendedPictographic letting us
+  //                    know that we are in an emoji sequence so our initial
+  //                    break question is answered as NO.
+  func checkIfInEmojiSequence(_ state: inout State) {
```

Comment:
But is State updated here, or can it be copied in?

Reply:
State's `isInEmojiSequence` gets updated here.